src/contracts/Input/ParsingDispatcher.php

```diff
@@ -100,16 +100,23 @@ protected function internalParse(array $data, string $mediaType)
     {
         list($mediaType, $version) = $this->parseMediaTypeVersion($mediaType);
 
+        $mediaTypeWithoutVersion = $this->fetchMediaTypeWithoutVersion($mediaType);
```

Comment:
According to `parseMediaTypeVersion` declaration (and judging how the code behaves) that method already separates media type and it's version.

Why it wasn't reused, even if you introduce a new public method to expose this functionality?